### ISSUE TITLE
[ENH]: Support for `\boldsymbol`

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#25643]
### Problem

Once #25359 is merged, it should be rather straightforward to add support for `\boldsymbol` which basically does the same thing as `\mathbfit` but is maybe a more standard solution.

### Proposed solution

Some inspiration can be taken from #22173 on how to add a new parsing element, although `QuotedString` should (most likely) not be used.

**Comment by devRD** [comment]
Hi @oscargus,

I have a working patch for `boldsymbol` in my local repository. Since #25359  is a dependency, I am unsure about opening a new PR.

Also, for generating the tests for `boldsymbol`, does it need `all` tests for the `baseline_images`, or similar to #22173 adding a lightweight test is fine?

**Comment by oscargus** [comment]
Great!

I think a lightweight test is enough.

Hopefully, #25359 is merged soon, so maybe one can wait for a bit before adding it? It is possible to open a new PR as well and just state that it depends on #25359.

**Comment by ksunden** [comment]
Xref #1366 

Discussion there seems to have centered around "do we wish to include amsmath things in the `mathtext` implementation or keep it 'pure latex', for simplicity and scoping".

That said, the barrier for `amsmath` is breached by `\genfrac` and `\binom` already, so that could be an argument for including `\boldsymbol`

However, the behavior of `\boldsymbol` is not quite the same as `bfit` as noted by @devrd [here](https://github.com/matplotlib/matplotlib/pull/25661#issuecomment-1532055435).

My personal prior would be that if `\boldsymbol` is included, it should behave as close to the latex behavior as possible, and thus is not as simple as aliasing `bfit`, though the implementation may mirror `bfit`.

**Comment by oscargus** [comment]
There are more things that are from ams and other packages (especially symbols). I'd say add what is useful.
